File: javascript/071_matching-brackets/matching-brackets.js
export const isPaired = (input) => {
  const mapa = {'(':')', '{':'}', '[':']'};
  const stack = [];
  for (let c of input) {
    if(c in mapa)
      stack.push(c);
    else if(Object.values(mapa).includes(c)) {
      if(mapa[stack.at(-1)] == c)
        stack.pop();
      else
        return false;
    }      
  }
  return !stack.length;
};

File: python/ex_053_matching-brackets/matching_brackets.py
pairs = {'(':')', '[':']', '{':'}'}

def is_paired(input_string):
    stack = []
    for char in input_string:
        if char in ')]}' and (not len(stack) or char != stack.pop()):
            return False
        if char in '([{':
            stack.append(pairs[char])
    return len(stack) == 0